test/prod/ProdTest.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import {SafeTransferLib, ERC20} from "@rari-capital/solmate/src/utils/SafeTransferLib.sol";
+
+import {PercentageMath} from "@morpho-dao/morpho-utils/math/PercentageMath.sol";
+import {Math} from "@morpho-dao/morpho-utils/math/Math.sol";
+import {WadRayMath} from "@morpho-dao/morpho-utils/math/WadRayMath.sol";
+
+import {BaseConfig} from "config/BaseConfig.sol";
+import "@forge-std/console.sol";
+import "@forge-std/Test.sol";
+
+contract ProdTest is Test, BaseConfig {
+    // Needed because AAVE packs the balance struct.
+    function dealAave(address who, uint104 amount) public {
+        // The slot of the balance struct "_balances" is 0.
+        bytes32 slot = keccak256(abi.encode(who, uint256(0)));
+        bytes32 initialValue = vm.load(aave, slot);
+        // The balance is stored in the first 104 bits.
+        bytes32 finalValue = ((initialValue >> 104) << 104) | bytes32(uint256(amount));
+        vm.store(aave, slot, finalValue);
+        require(ERC20(aave).balanceOf(who) == uint256(amount));
+    }
```

Comment:
This was also handled in aaveV3 optimizer this way: https://github.com/morpho-org/morpho-aavev3-optimizer/blob/5acc62c04a5e8032d4cce399dea31442ef6c2ad2/test/helpers/ForkTest.sol#L188-L198

At some point (I don't remember where), we even overridden forge-std's `deal` function to take into account the AAVE edge case (i really like this design)

I don't think creating a `ProdTest` layer for this is interesting: it adds a layer whereas we know that all tests in this folder are production tests (they all inherit from `prod/TestSetup`)

Reply:
> This was also handled in aaveV3 optimizer this way: https://github.com/morpho-org/morpho-aavev3-optimizer/blob/5acc62c04a5e8032d4cce399dea31442ef6c2ad2/test/helpers/ForkTest.sol#L188-L198

Interesting, I did not know that. Do you think we should change the code to match the one you linked ?

> I don't think creating a ProdTest layer for this is interesting: it adds a layer whereas we know that all tests in this folder are production tests (they all inherit from prod/TestSetup)

This is what I had initially in the first few commits of this PR, see [this commit](https://github.com/morpho-org/morpho-optimizers/pull/1669/commits/738ad6d71d05cd95c6330c8f13890ed66b979233), but then I realized that there are 2 different `TestSetup`: one for `aave-v2` and one for `compound`. So it would mean duplicating the function, do you think we should do it ?

Reply:
> Do you think we should change the code to match the one you linked ?

If your code works, I don't have a strong argument in favor of mine ^^ so keeping yours is fine

> So it would mean duplicating the function, do you think we should do it ?

Ah right, missed this! `ProdTest` is inherited by `TestSetup`. Well then i would only argue towards overriding forge-std's `deal` as in the code i referenced, keeping your workaround or mine (i don't care) ; so we don't have to check if the token is AAVE outside of this function

It can be useful when fuzzing the token but perhaps we don't do it in this repo. In which case let's just keep your version because this code is getting archived anyway

Reply:
Ok I changed both because they seem cleaner. One little caveat is that the AAVE token is not supposed to work with values greater than `type(uint104).max`, so we would have to distinguish the cases in fuzzing still (with the current solution at least)